code/modules/food_and_drinks/kitchen_machinery/grill_new.dm

```diff
@@ -43,3 +43,20 @@
 	for(var/obj/item/weapon/stock_parts/micro_laser/M in component_parts)
 		E += M.rating
 	efficiency = round((E/2), 1) // There's 2 lasers, so halve the effect on the efficiency to keep it balanced
+
+/obj/machinery/kitchen_machine/grill/special_attack(obj/item/weapon/grab/G, mob/user)
+	if(ishuman(G.affecting))
+		if(G.state < GRAB_AGGRESSIVE)
+			to_chat(user, "<span class='warning'>You need a better grip to do that!</span>")
+			return 1
+		var/mob/living/carbon/human/C = G.affecting
+		C.visible_message("<span class='danger'>[user] forces [C] onto [src], searing [C]'s body!</span>", \
+						"<span class='userdanger'>[user] forces you onto [src]! It burns!</span>")
+		C.emote("scream")
+		user.changeNext_move(CLICK_CD_MELEE)
+		C.adjustFireLoss(30)
+		add_logs(user, G.affecting, "burns on a grill")
+		qdel(G) //Removes the grip to prevent rapid sears and give you a chance to run
+		return 1
+	else
```

Comment:
You don't need the else here to properly `return 0`; that should just be the default behavior.